instrumentation/oshi/library/src/main/java/io/opentelemetry/instrumentation/oshi/JavaMetrics.java

```diff
@@ -0,0 +1,82 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.oshi;
+
+import io.opentelemetry.OpenTelemetry;
+import io.opentelemetry.common.Labels;
+import io.opentelemetry.metrics.AsynchronousInstrument.Callback;
+import io.opentelemetry.metrics.AsynchronousInstrument.DoubleResult;
+import io.opentelemetry.metrics.AsynchronousInstrument.LongResult;
+import io.opentelemetry.metrics.Meter;
+import java.lang.management.GarbageCollectorMXBean;
+import java.lang.management.ManagementFactory;
+import oshi.SystemInfo;
+import oshi.software.os.OSProcess;
+import oshi.software.os.OperatingSystem;
+
+/** Java Runtime Metrics Utility */
+public class JavaMetrics {
+  private static final String TYPE_LABEL_KEY = "type";
+
+  private JavaMetrics() {}
+
+  /** Register observers for java runtime metrics */
+  public static void registerObservers() {
+    Meter meter = OpenTelemetry.getMeterProvider().get(JavaMetrics.class.getName());
+    SystemInfo systemInfo = new SystemInfo();
+    OperatingSystem osInfo = systemInfo.getOperatingSystem();
+    OSProcess processInfo = osInfo.getProcess(osInfo.getProcessId());
+
+    meter
+        .longUpDownSumObserverBuilder("runtime.java.memory")
+        .setDescription("Runtime Java memory")
+        .setUnit("bytes")
+        .build()
+        .setCallback(
+            new Callback<LongResult>() {
+              @Override
+              public void update(LongResult r) {
+                processInfo.updateAttributes();
+                r.observe(processInfo.getResidentSetSize(), Labels.of(TYPE_LABEL_KEY, "rss"));
+                r.observe(processInfo.getVirtualSize(), Labels.of(TYPE_LABEL_KEY, "vms"));
+              }
+            });
+
+    meter
+        .doubleValueObserverBuilder("runtime.java.cpu_time")
+        .setDescription("Runtime Java CPU time")
+        .setUnit("seconds")
+        .build()
+        .setCallback(
+            new Callback<DoubleResult>() {
+              @Override
+              public void update(DoubleResult r) {
+                processInfo.updateAttributes();
+                r.observe(processInfo.getUserTime() * 1000, Labels.of(TYPE_LABEL_KEY, "user"));
+                r.observe(processInfo.getKernelTime() * 1000, Labels.of(TYPE_LABEL_KEY, "system"));
+              }
+            });
+
+    meter
+        .longValueObserverBuilder("runtime.java.gc_count")
+        .setDescription("Runtime Java GC count")
+        .setUnit("counts")
+        .build()
+        .setCallback(
+            new Callback<LongResult>() {
+              @Override
+              public void update(LongResult r) {
+                long gcCount = 0;
+                for (final GarbageCollectorMXBean gcBean :
+                    ManagementFactory.getGarbageCollectorMXBeans()) {
+                  gcCount += gcBean.getCollectionCount();
+                }
+
+                r.observe(gcCount, Labels.of(TYPE_LABEL_KEY, "count"));
+              }
+            });
```

Comment:
can this be part of the other (future) module that only depends on JVM-provided metrics?

Reply:
It depends on oshi.
If I will move it to separate module it will have the same dependency on oshi as system metrics.
Activation will be the same: presence of oshi dependency.
Therefore I doubt to split it to two modules.

Reply:
`runtime.java.gc_count` metric has dependency on oshi?

Reply:
oh, you mean only this specific metric, I thought you are talking about all jvm metrics here.

Reply:
I can create separate `instrumentation/jvm/libarary` with only this metric.

Reply:
👍 how about removing `runtime.java.gc_count` from this PR and then we'll merge this (sorry again it took so long!), and u can send separate PR with `instrumentation/jvm/library`?

Reply:
removed